sdks/python/apache_beam/transforms/combiners_test.py

```diff
@@ -399,6 +432,108 @@ def test_global_fanout(self):
+    assert_that(concat_per_key, equal_to(expected_concat_per_key),
+                label='concat per key')
+
+    # Verify the values of metrics are correct.
+    word_counter_filter = MetricsFilter().with_name('word_counter')
+    query_result = result.metrics().query(word_counter_filter)
+    if query_result['counters']:
+      word_counter = query_result['counters'][0]
+      self.assertEqual(word_counter.result, 5)
+
+    word_lengths_filter = MetricsFilter().with_name('word_lengths')
+    query_result = result.metrics().query(word_lengths_filter)
+    if query_result['counters']:
+      word_lengths = query_result['counters'][0]
+      self.assertEqual(word_lengths.result, 16)
+
+    word_len_dist_filter = MetricsFilter().with_name('word_len_dist')
+    query_result = result.metrics().query(word_len_dist_filter)
+    if query_result['distributions']:
+      word_len_dist = query_result['distributions'][0]
+      self.assertEqual(word_len_dist.result.mean, 3.2)
+
+    last_word_len_filter = MetricsFilter().with_name('last_word_len')
+    query_result = result.metrics().query(last_word_len_filter)
+    if query_result['gauges']:
+      last_word_len = query_result['gauges'][0]
+      self.assertEqual(last_word_len.result.value, 6)
+
+  # Test that three different kinds of metrics work with the customized
+  # CounterIncrememtingCombineFn when the PCollection is empty.
+  def test_simple_combine_empty(self):
+    p = TestPipeline()
+    input = p | beam.Create([])
+
+    # The result of concatenating all values regardless of key.
+    global_concat = (input
+                     | beam.Values()
+                     | beam.CombineGlobally(CounterIncrememtingCombineFn()))
+
+    # The (key, concatenated_string) pairs for all keys.
+    concat_per_key = (input | beam.CombinePerKey(
+        CounterIncrememtingCombineFn()))
+
+    result = p.run()
+    result.wait_until_finish()
+
+    # Verify the concatenated strings are correct.
+    assert_that(global_concat, equal_to(['']), label='global concat')
+    assert_that(concat_per_key, equal_to([]), label='concat per key')
+
+    # Verify the values of metrics are correct.
+    word_counter_filter = MetricsFilter().with_name('word_counter')
+    query_result = result.metrics().query(word_counter_filter)
+    if query_result['counters']:
+      word_counter = query_result['counters'][0]
+      self.assertEqual(word_counter.result, 0)
+
+    word_lengths_filter = MetricsFilter().with_name('word_lengths')
+    query_result = result.metrics().query(word_lengths_filter)
+    if query_result['counters']:
+      word_lengths = query_result['counters'][0]
+      self.assertEqual(word_lengths.result, 0)
+
+    word_len_dist_filter = MetricsFilter().with_name('word_len_dist')
+    query_result = result.metrics().query(word_len_dist_filter)
+    if query_result['distributions']:
+      word_len_dist = query_result['distributions'][0]
+      self.assertEqual(word_len_dist.result.mean, 0)
```

Comment:
The mean should not be 0, it should be undefined. You could test the count perhaps.

Reply:
Done. Changed to 'count'.